File: dotnet/Lexy/Lexy.Compiler/Language/Scenarios/ScenarioParameters.cs
using System.Collections.Generic;
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Scenarios
{
    public class ScenarioParameters : ParsableNode
    {
        public IList<AssignmentDefinition> Assignments { get; } = new List<AssignmentDefinition>();

        public ScenarioParameters(SourceReference reference) : base(reference)
        {
        }

        public override IParsableNode Parse(IParserContext context)
        {
            var assignment = AssignmentDefinition.Parse(context);
            if (assignment != null)
            {
                Assignments.Add(assignment);
            }
            return this;
        }

        public override IEnumerable<INode> GetChildren() => Assignments;

        protected override void Validate(IValidationContext context)
        {
        }
    }
}

File: typescipt/src/language/scenarios/scenarioParameters.ts


export class ScenarioParameters extends ParsableNode {
   public Array<AssignmentDefinition> Assignments = list<AssignmentDefinition>(): new;

   public ScenarioParameters(SourceReference reference) {
     super(reference);
   }

   public override parse(context: IParseLineContext): IParsableNode {
     let assignment = AssignmentDefinition.parse(context);
     if (assignment != null) Assignments.Add(assignment);
     return this;
   }

   public override getChildren(): Array<INode> {
     return Assignments;
   }

   protected override validate(context: IValidationContext): void {
   }
}
